### ISSUE TITLE
Fan Auto and Turbo mode switches

### ISSUE THREAD

**Issue by leandromagro** [kovapatrik/homebridge-midea-platform#102]
**Is your feature request related to a problem? Please describe:**

Fan Auto and Turbo mode switches request

**Describe the solution you'd like:**

Since the automations possibilities are endless, having the Fan Auto and Turbo mode as switches in Home app, will allow to create a new set of automation rules. 

**Describe alternatives you've considered:**

none

**Additional context:**

none. If needed ping me a message.

**Comment by kovapatrik** [comment]
Fan auto mode is accessible using the fan accessory switch. 

What is turbo mode? I don't see such option for my AC. Can you please turn on debug mode in your Homebridge instance, enable verbose logging for one of your device, then turn on and off turbo mode using the official app and send me the logs?

**Comment by leandromagro** [comment]
About the Fan mode - my bad. I didn't noticed that. Thanks.

about the log: (I believe what we area looking for is the boost mode - these translations kills me)
[20/08/2024, 11:04:34] [midea-platform] [Suíte] Value for POWER changed from 'false' to 'true'
[20/08/2024, 11:04:34] [midea-platform] [Suíte] Value for BOOST_MODE changed from 'false' to 'true'
[20/08/2024, 11:04:34] [midea-platform] [Suíte] Set attribute POWER to: true
[20/08/2024, 11:04:34] [midea-platform] [Suíte] Set attribute BOOST_MODE to: true
[20/08/2024, 11:04:34] [midea-platform] [Suíte] Attempt to set unsupported attribute BOOST_MODE to true
[20/08/2024, 11:04:37] [midea-platform] [Suíte] Value for OUTDOOR_TEMPERATURE changed from 'undefined' to '31.3'
[20/08/2024, 11:04:37] [midea-platform] [Suíte] Value for INDOOR_HUMIDITY changed from 'undefined' to '0'
[20/08/2024, 11:04:37] [midea-platform] [Suíte] Set attribute OUTDOOR_TEMPERATURE to: 31.3
[20/08/2024, 11:04:37] [midea-platform] [Suíte] Set attribute INDOOR_HUMIDITY to: 0
[20/08/2024, 11:04:37] [midea-platform] [Suíte] Attempt to set unsupported attribute INDOOR_HUMIDITY to 0
[20/08/2024, 11:04:40] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:04:43] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:04:46] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:04:51] [midea-platform] [Suíte] Value for BOOST_MODE changed from 'true' to 'false'
[20/08/2024, 11:04:51] [midea-platform] [Suíte] Set attribute BOOST_MODE to: false
[20/08/2024, 11:04:51] [midea-platform] [Suíte] Attempt to set unsupported attribute BOOST_MODE to false
[20/08/2024, 11:04:54] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:04:57] [midea-platform] [Suíte] Refreshing status...
[20/08/2024, 11:04:57] [midea-platform] [Suíte] Value for POWER changed from 'true' to 'false'
[20/08/2024, 11:04:57] [midea-platform] [Suíte] Set attribute POWER to: false
[20/08/2024, 11:04:57] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:04:57] [midea-platform] [Closet] Refreshing status...
[20/08/2024, 11:04:57] [midea-platform] [Escritório] Refreshing status...
[20/08/2024, 11:04:57] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:04:58] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:04:58] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:05:00] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:05:00] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:05:01] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:05:04] [midea-platform] [Suíte] Status unchanged

**Comment by kovapatrik** [comment]
Yes, that's what I needed, thank you!

**Comment by kovapatrik** [comment]
I've added the possibility of doing this with `v1.1.2-beta.6`. Can you try it, please? Also, the switch should turn on the device, if it was off before, but turning off the switch will not power it down. If you think this logic is incorrect, please le me know.

**Comment by leandromagro** [comment]
About the Fan. I have tested here and it show the slider. There is an option, when you set the AC to cool or heat, to set the fan to auto. I believe this is the parameter: [20/08/2024, 11:17:22] [midea-platform] [Escritório] Value for FAN_AUTO changed from 'true' to 'false'
Is this FAN_AUTO could be a switch?

[20/08/2024, 11:15:50] [midea-platform] [Escritório] Refreshing status...
[20/08/2024, 11:15:51] [midea-platform] [Suíte] Refreshing status...
[20/08/2024, 11:15:51] [midea-platform] [Suíte] Value for INDOOR_TEMPERATURE changed from '25.2' to '25.4'
[20/08/2024, 11:15:51] [midea-platform] [Suíte] Set attribute INDOOR_TEMPERATURE to: 25.4
[20/08/2024, 11:15:51] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:15:51] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:15:51] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:15:51] [midea-platform] [Closet] Refreshing status...
[20/08/2024, 11:15:52] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:15:53] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:15:54] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:16:21] [midea-platform] [Escritório] Refreshing status...
[20/08/2024, 11:16:21] [midea-platform] [Suíte] Refreshing status...
[20/08/2024, 11:16:21] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:16:21] [midea-platform] [Suíte] Value for SCREEN_DISPLAY changed from 'false' to 'true'
[20/08/2024, 11:16:21] [midea-platform] [Suíte] Value for SCREEN_DISPLAY_NEW changed from 'false' to 'true'
[20/08/2024, 11:16:21] [midea-platform] [Suíte] Set attribute SCREEN_DISPLAY to: true
[20/08/2024, 11:16:21] [midea-platform] [Suíte] Set attribute SCREEN_DISPLAY_NEW to: true
[20/08/2024, 11:16:21] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:16:22] [midea-platform] [Closet] Refreshing status...
[20/08/2024, 11:16:22] [midea-platform] [Escritório] Value for MODE changed from '1' to '2'
[20/08/2024, 11:16:22] [midea-platform] [Escritório] Set attribute MODE to: 2
[20/08/2024, 11:16:22] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:16:24] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:16:24] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:16:51] [midea-platform] [Escritório] Refreshing status...
[20/08/2024, 11:16:51] [midea-platform] [Suíte] Refreshing status...
[20/08/2024, 11:16:52] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:16:52] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:16:52] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:16:52] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:16:52] [midea-platform] [Closet] Refreshing status...
[20/08/2024, 11:16:53] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:16:53] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:16:54] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:17:21] [midea-platform] [Escritório] Refreshing status...
[20/08/2024, 11:17:22] [midea-platform] [Suíte] Refreshing status...
[20/08/2024, 11:17:22] [midea-platform] [Escritório] Value for FAN_SPEED changed from '102' to '60'
[20/08/2024, 11:17:22] [midea-platform] [Escritório] Value for FAN_AUTO changed from 'true' to 'false'
[20/08/2024, 11:17:22] [midea-platform] [Escritório] Set attribute FAN_SPEED to: 60
[20/08/2024, 11:17:22] [midea-platform] [Escritório] Set attribute FAN_AUTO to: false
[20/08/2024, 11:17:22] [midea-platform] [Suíte] Value for INDOOR_TEMPERATURE changed from '25.4' to '25.3'
[20/08/2024, 11:17:22] [midea-platform] [Suíte] Set attribute INDOOR_TEMPERATURE to: 25.3
[20/08/2024, 11:17:22] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:17:22] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:17:22] [midea-platform] [Closet] Refreshing status...
[20/08/2024, 11:17:23] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:17:24] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:17:25] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:17:52] [midea-platform] [Escritório] Refreshing status...
[20/08/2024, 11:17:52] [midea-platform] [Escritório] Value for FAN_SPEED changed from '60' to '102'
[20/08/2024, 11:17:52] [midea-platform] [Escritório] Value for FAN_AUTO changed from 'false' to 'true'
[20/08/2024, 11:17:52] [midea-platform] [Escritório] Set attribute FAN_SPEED to: 102
[20/08/2024, 11:17:52] [midea-platform] [Escritório] Set attribute FAN_AUTO to: true
[20/08/2024, 11:17:52] [midea-platform] [Suíte] Refreshing status...
[20/08/2024, 11:17:53] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:17:53] [midea-platform] [Closet] Refreshing status...
[20/08/2024, 11:17:53] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:17:53] [midea-platform] [Suíte] Status unchanged
[20/08/2024, 11:17:53] [midea-platform] [Closet] Status unchanged
[20/08/2024, 11:17:54] [midea-platform] [Escritório] Status unchanged
[20/08/2024, 11:17:55] [midea-platform] [Closet] Status unchanged

**Comment by leandromagro** [comment]
> I've added the possibility of doing this with `v1.1.2-beta.6`. Can you try it, please? Also, the switch should turn on the device, if it was off before, but turning off the switch will not power it down. If you think this logic is incorrect, please le me know.

I tested it and it is perfect. I think the logic is also good mainly if you think for an automation to cool /heat a room as fast as it can. So you don't need to send a lot of parameters, just turn this switch on.
Tested: 
AC off - switch on - result: turned on the AC in the boost mode
AC on - switch off -> on - result: turned on the boost mode
AC on - switch on -> off - result: turned off the boost mode but not the AC.
